examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -72,3 +72,137 @@ bool RvcServiceAreaDelegate::HandleSkipCurrentLocation(std::string & skipStatusT
 
     return ret_value;
 };
+
+
+
+//*************************************************************************
+// Supported Locations accessors
+
+void RvcServiceAreaDelegate::ReloadSupportedLocationsVector()
+{
+    mSupportedLocationsByIndex.resize(mSupportedLocationsById.size());
+
+    uint32_t locationIndex = 0;
+    for (auto & entry : mSupportedLocationsById)
+    {
+        mSupportedLocationsByIndex[locationIndex++] = &entry.second;
+    }
+}
+
+size_t RvcServiceAreaDelegate::GetNumberOfSupportedLocations() 
+{
+    return mSupportedLocationsByIndex.size();
+}
+
+bool RvcServiceAreaDelegate::RvcServiceAreaDelegate::GetSupportedLocationByIndex(uint32_t listIndex, const LocationStructureWrapper *& supportedLocation) 
+{
+    bool ret_value = false;
+
+    if (listIndex < mSupportedLocationsByIndex.size())
+    {
+        supportedLocation = mSupportedLocationsByIndex[listIndex];
+        ret_value = true;
+    }
+
+    return ret_value;
+};
+
+bool RvcServiceAreaDelegate::GetSupportedLocationById(uint32_t aLocationId, const LocationStructureWrapper *& supportedLocation)
+{
+    bool ret_value = false;
+
+    auto locationIter = mSupportedLocationsById.find(aLocationId);
```

Comment:
To expand on a comment I made earlier, implementing this method is easy in this case since we have the nice functionality of `find`. On an embedded target this wouldn't necessarily be available, so forcing users to implement such methods is not ideal. Especially since we can implement this functionality in the SDK just from the top 2 methods.